File: common/src/main/java/dev/number6/db/adaptor/DatabaseServiceConfigurationAdaptor.java
package dev.number6.db.adaptor;

import dev.number6.db.port.DatabaseConfigurationPort;
import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapperConfig;

public class DatabaseServiceConfigurationAdaptor implements DatabaseConfigurationPort {

    public static final String DB_TABLE = "DB_TABLE";
    private final DynamoDBMapperConfig dynamoDBMapperConfig;

    public DatabaseServiceConfigurationAdaptor() {
        dynamoDBMapperConfig = new DynamoDBMapperConfig.TableNameOverride(System.getenv(DB_TABLE)).config();
    }

    @Override
    public DynamoDBMapperConfig getDynamoDBMapperConfig() {
        return dynamoDBMapperConfig;
    }
}

File: common/src/main/kotlin/dev/number6/db/adaptor/DatabaseServiceConfigurationAdaptor.kt
package dev.number6.db.adaptor

import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapperConfig
import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapperConfig.TableNameOverride
import dev.number6.db.port.DatabaseConfigurationPort

class DatabaseServiceConfigurationAdaptor : DatabaseConfigurationPort {
    override val dynamoDBMapperConfig: DynamoDBMapperConfig by lazy {
        TableNameOverride(System.getenv(DB_TABLE)).config()
    }

    companion object {
        const val DB_TABLE = "DB_TABLE"
    }
}